test/e2e/network/netpol/test_helper.go

```diff
@@ -0,0 +1,145 @@
+		testCase.Reachability.PrintSummary(true, true, true)
+		framework.Failf("Had %d wrong results in reachability matrix", wrong)
+	} 
+	if isVerbose {
+			testCase.Reachability.PrintSummary(true, true, true)
+	}
+	framework.Logf("VALIDATION SUCCESSFUL")
+
+}
+
+// UpdateNamespaceLabels sets the labels for a namespace
+func UpdateNamespaceLabels(k8s *Kubernetes, ns string, newNsLabel map[string]string) {
+	err := k8s.setNamespaceLabels(ns, newNsLabel)
+	framework.ExpectNoError(err, "Update namespace %s labels", ns)
+	err = wait.PollImmediate(1*time.Second, 30*time.Second, func() (done bool, err error) {
+		namespace, err := k8s.getNamespace(ns)
+		if err != nil {
+			return false, err
+		}
+		for key, expected := range newNsLabel {
+			if actual, ok := namespace.Labels[key]; !ok || (expected != actual) {
+				return false, nil
+			}
+		}
+		return true, nil
+	})
+	framework.ExpectNoError(err, "Unable to wait for ns %s to update labels", ns)
+}
+
+// AddPodLabels adds new labels to a deployment's template
+func AddPodLabels(k8s *Kubernetes, pod *Pod, newPodLabels map[string]string) {
+	kubePod, err := k8s.ClientSet.CoreV1().Pods(pod.Namespace).Get(context.TODO(), pod.Name, metav1.GetOptions{})
+	framework.ExpectNoError(err, "Failing to get pod %s/%s", pod.Namespace, pod.Name)
+	for key, val := range newPodLabels {
+		kubePod.Labels[key] = val
+	}
+	_, err = k8s.ClientSet.CoreV1().Pods(pod.Namespace).Update(context.TODO(), kubePod, metav1.UpdateOptions{})
+	framework.ExpectNoError(err, "Failing to add pod %s/%s labels", pod.Namespace, pod.Name)
+
+	err = wait.PollImmediate(1*time.Second, 30*time.Second, func() (done bool, err error) {
+		waitForPod, err := k8s.GetPod(pod.Namespace, pod.Name)
+		if err != nil {
+			return false, err
+		}
+		for key, expected := range newPodLabels {
+			if actual, ok := waitForPod.Labels[key]; !ok || (expected != actual) {
+				return false, nil
+			}
+		}
+		return true, nil
+	})
+	framework.ExpectNoError(err, "Unable to wait for pod %s/%s to update labels", pod.Namespace, pod.Name)
+}
+
+// ResetNamespaceLabels resets the labels for a namespace
+func ResetNamespaceLabels(k8s *Kubernetes, ns string) {
+	UpdateNamespaceLabels(k8s, ns, (&Namespace{Name: ns}).LabelSelector())
+}
+
+// ResetPodLabels resets the labels for a deployment's template
+func ResetPodLabels(k8s *Kubernetes, pod *Pod) {
+	kubePod, err := k8s.ClientSet.CoreV1().Pods(pod.Namespace).Get(context.TODO(), pod.Name, metav1.GetOptions{})
+	framework.ExpectNoError(err, "Failing to get pod %s/%s", pod.Namespace, pod.Name)
+	kubePod.Labels = pod.LabelSelector()
+	_, err = k8s.ClientSet.CoreV1().Pods(pod.Namespace).Update(context.TODO(), kubePod, metav1.UpdateOptions{})
+	framework.ExpectNoError(err, "Failing to add pod %s/%s labels", pod.Namespace, pod.Name)
+
+	err = wait.PollImmediate(1*time.Second, 30*time.Second, func() (done bool, err error) {
+		waitForPod, err := k8s.GetPod(pod.Namespace, pod.Name)
+		if err != nil {
+			return false, err
```

Comment:
do we want to fail or do we want to retry?
if the later, ` return false, nil`

Reply:
@mattfenwick  can you make the call on this one ?

Reply:
Since we're updating the pod, not creating it: we should be able to get it. If we can't, that's a failure.

^^ that's my thought process

Reply:
@aojea strong opinion on that one ? or keep for now and iterate post merge if needed?

Reply:
I'd say `return false, nil`, the GetPod is an apiserver GET request that can go wrong by different causes unrelated to the test itself (I saw a lot of these in busy CI: slow etcd, constrained network, ...) not necessarily because the pod does not exist ... 
you can look for the type of apiserver error, but if the pod wasn't working will not be able to get it in the next iterations and is not worth the effort

Reply:
ok, fixed